### DISCUSSION TITLE
Watermark on word document body

### DISCUSSION THREAD

**Discussion by SamiRND** [OfficeDev/Open-XML-SDK#1849] [General]
I am using open XML 3.2 version to add watermark on word file. I am using the below code. 

```csharp
 using (WordprocessingDocument wordDocument = WordprocessingDocument.Open(filePath, true))
 {


     // Get the body of the document
     var body = wordDocument.MainDocumentPart.Document.Body;

     // Check if the document body is null or corrupted, and rebuild it if necessary
     if (body == null)
     {
         body = new Body(); // Create a new body if needed
         wordDocument.MainDocumentPart.Document.Append(body);
     }

   
     body.AppendChild(new DocumentFormat.OpenXml.Drawing.Paragraph(new DocumentFormat.OpenXml.Drawing.Run(new DocumentFormat.OpenXml.Drawing.Text("This is some inline text"))));

     // Ensure all parts are saved correctly
     wordDocument.MainDocumentPart.Document.Save();
}
```

There is no error. but the word file gets corrupted and is not opening.

![image](https://github.com/user-attachments/assets/b0534376-1d31-4a32-b946-c80c909c2eab)

error message i am getting when open the file.

**Comment by mikeebowen** [comment]
Hi @SamiRND,

You're using the wrong namespace for `Paragraph`, `Run`, and `Text`. You need to use the `DocumentFormat.OpenXml.Wordprocessing` namespace.

```csharp
new DocumentFormat.OpenXml.Wordprocessing.Paragraph(new DocumentFormat.OpenXml.Wordprocessing.Run(new DocumentFormat.OpenXml.Wordprocessing.Text("Hello World")))
```

**Comment by mikeebowen** [comment]
Hi @SamiRND, I haven't heard from you, so I will close this discussion as answered. If the above solution doesn't work for you, please reopen the discussion and tag me.
